test/e2e/operator_test.go

```diff
@@ -37,3 +44,79 @@ func TestOperandImageVersion(t *testing.T) {
 	}
 	require.Fail(t, "operator kube-apiserver image version not found")
 }
+
+func TestRevisionLimits(t *testing.T) {
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	operatorConfigClient, err := operatorversionedclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	operatorConfigInformers := operatorv1informers.NewSharedInformerFactory(operatorConfigClient, 10*time.Minute)
+	operatorClient := &operatorclient.OperatorClient{
+		Informers: operatorConfigInformers,
+		Client:    operatorConfigClient.OperatorV1(),
+	}
+
+	// Get current revision limits
+	operatorSpec, _, _, err := operatorClient.GetStaticPodOperatorState()
+	totalRevisionLimit := operatorSpec.SucceededRevisionLimit + operatorSpec.FailedRevisionLimit
+	if operatorSpec.SucceededRevisionLimit == 0 {
+		totalRevisionLimit += 5
+	}
+	if operatorSpec.FailedRevisionLimit == 0 {
+		totalRevisionLimit += 5
+	}
+
+	revisions, err := getRevisionStatuses(kubeClient)
+	require.NoError(t, err)
+
+	// Check if revisions are being quickly created to test for operator hotlooping
+	changes := 0
+	pollsWithoutChanges := 0
+	lastRevisionCount := len(revisions)
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
Why don't we just read the kubeapiserver resource and see if the number is greater than 40.  If so, we're probably hot looping?

Reply:
You mean `LatestAvailableRevision`? That might not update quick enough if we're creating a ton of revisions, but I think the configmaps will. I can just check those for a revision count instead of this loop